cfgrammar/src/lib/newlinecache.rs

```diff
@@ -11,6 +11,14 @@ impl NewlineCache {
         }
     }
 
+    /// Construct a `NewlineCache` directly from a `&str`. This is equivalent to creating a blank
+    /// `NewlineCache` and [`feed()`]ing the string directly in.
+    pub fn from_str(s: &str) -> Self {
```

Comment:
Clippy is going to want this to implement,
https://doc.rust-lang.org/stable/std/str/trait.FromStr.html

But that returns a `Result`, we could use `type Error = std::convert::Infallible;`
I'm not sure how that works/whether it e.g. allows use to use `?` in functions that don't return results though.

Another option would be `#[allow(clippy::should_implement_trait)]` on this function, or a name like `cache_str`?
I'm not sure about allowing that lint globally though, if there may be cases we want it to catch.

Reply:
That's unfortunate. My gut instinct is to just ignore Clippy for this specific function but let me chew on it.

Reply:
Another option is that we could implement this using the `From` trait.

Reply:
I think `FromStr` is the least worst option. Fixed in 73f40a1.